Client/Frontend/Browser/Playlist/Controller/PlaylistViewController.swift

```diff
@@ -0,0 +1,831 @@
+            playerView.isOverlayDisplayed = true
+            
+            player.pause()
+        } else {
+            playerView.toggleOverlays(showOverlay: playerView.isOverlayDisplayed)
+        }
+    }
+    
+    func stop(_ videoView: VideoView) {
+        playerView.controlsView.playPauseButton.setImage(#imageLiteral(resourceName: "playlist_play"), for: .normal)
+        playerView.toggleOverlays(showOverlay: true)
+        playerView.isOverlayDisplayed = true
+        
+        player.stop()
+    }
+    
+    func seekBackwards(_ videoView: VideoView) {
+        player.seekBackwards()
+    }
+    
+    func seekForwards(_ videoView: VideoView) {
+        player.seekForwards()
+    }
+    
+    func seek(_ videoView: VideoView, to time: TimeInterval) {
+        player.seek(to: time)
+    }
+    
+    func seek(_ videoView: VideoView, relativeOffset: Float) {
+        if let currentItem = player.currentItem {
+            let seekTime = CMTimeMakeWithSeconds(Float64(CGFloat(relativeOffset) * CGFloat(currentItem.asset.duration.value) / CGFloat(currentItem.asset.duration.timescale)), preferredTimescale: currentItem.currentTime().timescale)
+            seek(videoView, to: seekTime.seconds)
+        }
+    }
+    
+    func setPlaybackRate(_ videoView: VideoView, rate: Float) {
+        player.setPlaybackRate(rate: rate)
+    }
+    
+    func togglePlayerGravity(_ videoView: VideoView) {
+        player.toggleGravity()
+    }
+    
+    func toggleRepeatMode(_ videoView: VideoView) {
+        player.toggleRepeatMode()
+    }
+    
+    func load(_ videoView: VideoView, url: URL, autoPlayEnabled: Bool) -> AnyPublisher<Void, Error> {
+        load(videoView, asset: AVURLAsset(url: url), autoPlayEnabled: autoPlayEnabled)
+    }
+    
+    func load(_ videoView: VideoView, asset: AVURLAsset, autoPlayEnabled: Bool) -> AnyPublisher<Void, Error> {
+        assetLoadingStateObservers.removeAll()
+        player.stop()
+        
+        return Future { [weak self] resolver in
+            guard let self = self else {
+                resolver(.failure("User Cancelled Playback"))
+                return
+            }
+            
+            self.player.load(asset: asset)
+            .receive(on: RunLoop.main)
+            .sink(receiveCompletion: { error in
+                if case .failure(let error) = error {
+                    resolver(.failure(error))
+                }
+            }, receiveValue: { [weak self] isNewItem in
+                guard let self = self else {
+                    resolver(.failure("User Cancelled Playback"))
+                    return
+                }
+                
+                guard let item = self.player.currentItem else {
+                    resolver(.failure("Couldn't load playlist item"))
+                    return
+                }
+                
+                // We are playing the same item again..
+                if !isNewItem {
```

Comment:
I have a feeling that some of code between car play and regular player can be shared, might be worth exploring what would be nice to share and if it even makes sense